transactor/transact.go

```diff
@@ -320,6 +321,18 @@ func (tx *Transaction) complete() {
 			logrus.Debugf("transactor(%d): abort succeeded", tx.ID)
 		}
 	}
+
+	// Update transaction record.
+	_, err = dal.SetTransaction(tx.Engine, &dal.Transaction{
+		ID:        tx.ID,
+		StartTime: tx.StartTime,
+		EndTime:   tx.EndTime,
+		Error:     tx.Error,
```

Comment:
Yes, the error is intended to be for the error during the transaction rather than a operational error (such as writing transaction itself).